extensions-core/protobuf-extensions/src/main/java/org/apache/druid/data/input/protobuf/ProtobufInputRowParser.java

```diff
@@ -37,39 +32,32 @@
 import org.apache.druid.data.input.MapBasedInputRow;
 import org.apache.druid.data.input.impl.JSONParseSpec;
 import org.apache.druid.data.input.impl.ParseSpec;
-import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.java.util.common.logger.Logger;
 import org.apache.druid.java.util.common.parsers.ParseException;
 import org.apache.druid.java.util.common.parsers.Parser;
 import org.apache.druid.utils.CollectionUtils;
 
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.MalformedURLException;
-import java.net.URL;
 import java.nio.ByteBuffer;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
 
 public class ProtobufInputRowParser implements ByteBufferInputRowParser
 {
+  private static final Logger LOG = new Logger(ByteBufferInputRowParser.class);
+
   private final ParseSpec parseSpec;
-  private final String descriptorFilePath;
-  private final String protoMessageType;
-  private Descriptor descriptor;
+  private final ProtobufBytesDecoder protobufBytesDecoder;
   private Parser<String, Object> parser;
   private final List<String> dimensions;
 
   @JsonCreator
   public ProtobufInputRowParser(
       @JsonProperty("parseSpec") ParseSpec parseSpec,
-      @JsonProperty("descriptor") String descriptorFilePath,
-      @JsonProperty("protoMessageType") String protoMessageType
+      @JsonProperty("protoBytesDecoder") ProtobufBytesDecoder protobufBytesDecoder
```

Comment:
So that this change is not incompatible with previous druid versions, I think we should probably consider leaving the old constructor properties in place and marking them as deprecated instead of removing them, and then enforce that only one of `protoBytesDecoder` or `descriptor`/`protoMessageType` can be set, and then automatically construct a 'FileBasedProtobufBytesDecoder' if the old properties are set and the new is null.

Then in a future version we can consider dropping them

Reply:
Good idea, I'll implement it.